### PR TITLE
Adding a way to compare binary files (bcfs)

### PR DESCRIPTION
**Pull request by markiskander**
Much like how bam_to_sam works, I added a bcf_to_vcf in order to compare binary files.

### CODE REVIEW

**Comment by davebx** [comment]
:+1: I had something very similar in a branch somewhere, but this is a cleaner solution.

File: test/functional/tools/compare_bcfs.xml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+<tool id="compare_bcfs" name="Compare BCFs" version ="0.0.1">
+  <description>Compare body of two BCF files</description>
+        <requirements>
+                <requirement type="package" version="1.2">bcftools</requirement>
```
Comment (bgruening) [inline test/functional/tools/compare_bcfs.xml]: Could you please adjust the indentation here.

**Comment by bgruening** [comment]
@markiskander this is really nice! Please adjust the indentation and we will merge this :)

**Comment by markiskander** [comment]
I suspect the failing CI is due to something unrelated to my changes? I might be wrong but it doesn't seem like those last 3 are getting very far before failing.

**Comment by martenson** [comment]
@galaxybot test this

**Comment by markiskander** [comment]
Alright guys I'll look into the failing test first thing Monday morning. Thanks!

**Comment by bgruening** [comment]
@martenson the tests passed before the last change. Can it be that bcftools disapproved from the test docker image?

**Comment by martenson** [comment]
@bgruening What tests do you mean? There is only one failing test and it seems related to this PR.

**Comment by markiskander** [comment]
Ok, so after some testing, it seems:
if attributes.get( 'ftype', None ) == 'bcf':

is never being satisfied, even though I have this in the XML:
http://pastebin.com/PqtU5re7

so I'm not sure why it doesn't 'detect' it as bcf?

'attributes' at this point is: 
{
  'sort': False, 
  'compare': 'diff', 
  'elements': {}, 
  'assert_list': None, 
  'lines_diff': 0, 
  'delta': 10000, 
  'primary_datasets': {}, 
  'md5': None, 
  'extra_files': [], 
  'metadata': {}
}

**Comment by bgruening** [comment]
@markiskander your output needs also to be bcf. You can also add `ftype="bcf"` to our output test dataset.

**Comment by markiskander** [comment]
Huh. Seems like check_commands isn't actually...executing the command? The expected output file isn't being generated with check_commands. I can't find it in the specified tmp directory (and I made sure to run the test with --no_cleanup

**Comment by markiskander** [comment]
Hey guys. So it seems bcf_to_vcf isn't really what I thought it was. It wouldn't really work for our uses. I was informed that internally, planemo should be able to handle bcf comparison just fine. So it seems we'll explore an alternate way of comparing bcf.gz, likely through a custom converter, etc. 

So PR is still there if you guys want to use it, but I don't really know how to write tests for it since it doesn't work the way I initially thought it did. Thanks @bgruening and @davebx for helping me on this!

**Comment by jmchilton** [comment]
Based on authors comments I am going to close this, but @davebx if you want to take this over and polish it up it seems promising to me.

**Pull request closed without merging**